tests/unit/test_utils.py

```diff
@@ -701,6 +704,31 @@ def test_manylinux_check_glibc_version(self):
                     # Didn't find the warning we were expecting
                     assert False
 
+    def test_glibc_version_string_confstr_fail(self, monkeypatch):
```

Comment:
How about a successful case, too? In addition, it would probably be good to have at least one successful test of `glibc_version_string()`, which you also modified. (The code for this test can be basically the same as the one for `glibc_version_string_confstr()`.)